### PR TITLE
Rename the provider from tmc to tanzu-mission-control

### PR DESCRIPTION
**Pull request by shobha2626**
1. **What this PR does / why we need it**:

Rename the provider from tmc => tanzu-mission-control

2. **Which issue(s) this PR fixes**

        (optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged):

        Fixes #

3. **Additional information**

4. **Special notes for your reviewer**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ already.
 The syntax is as follows:
 
 ```hcl
-provider "tmc" {
+provider "tanzu-mission-control" {
```
Comment (shreyassreenivas) [inline README.md:28]: Update it according the registry.terraform.io

File: README.md [hunk 2]
```diff
@@ -118,6 +118,10 @@ Run the command:
 $ make acc-test
 ```
 
+### Test provider changes locally
+Please make use of a unique path as provided in the `Makefile` while building the provider with changes 
+and kindly use the same path in the source while using the provider to test the local changes.
+
```
Comment (shreyassreenivas) [inline README.md:124]: Give the example of the provider init from the dev path

File: internal/resources/cluster/tkgservicevsphere/resource_tkg_service_vsphere.go [hunk 3]
```diff
@@ -16,7 +16,7 @@ import (
 
 var TkgServiceVsphere = &schema.Schema{
 	Type:        schema.TypeList,
-	Description: "The tkg service vsphere cluster spec",
+	Description: "The Tanzu Kubernetes Grid Service VSphere cluster spec",
```
Comment (shreyassreenivas) [inline internal/resources/cluster/tkgservicevsphere/resource_tkg_service_vsphere.go:19]: Tanzu Kubernetes Grid Service (tkgs)  cluster spec

File: internal/resources/cluster/tkgvsphere/resource_tkgvsphere.go [hunk 4]
```diff
@@ -16,7 +16,7 @@ import (
 
 var TkgVsphereClusterSpec = &schema.Schema{
 	Type:        schema.TypeList,
-	Description: "The TKG Vsphere cluster spec",
+	Description: "The Tanzu Kubernetes Grid Vsphere cluster spec",
```
Comment (shreyassreenivas) [inline internal/resources/cluster/tkgvsphere/resource_tkgvsphere.go:19]: mention the short form

File: README.md (hunk 1 above)
Comment (shobha2626) [reply README.md:37]: Done

File: README.md (hunk 2 above)
Comment (shobha2626) [reply README.md:132]: Done

File: internal/resources/cluster/tkgservicevsphere/resource_tkg_service_vsphere.go (hunk 3 above)
Comment (shobha2626) [reply internal/resources/cluster/tkgservicevsphere/resource_tkg_service_vsphere.go]: Done

File: internal/resources/cluster/tkgvsphere/resource_tkgvsphere.go (hunk 4 above)
Comment (shobha2626) [reply internal/resources/cluster/tkgvsphere/resource_tkgvsphere.go]: Done

File: internal/resources/cluster/constants.go [hunk 5]
```diff
@@ -16,8 +16,8 @@ const (
 	attachClusterKubeConfigKey  = "kubeconfig_file"
 	attachClusterKey            = "attach_k8s_cluster"
 	attachClusterDescriptionKey = "description"
-	waitKey                     = "ready_wait_timeout"
-	ResourceName                = "tmc_cluster"
+	waitKey                     = "wait_until_ready"
```
Comment (shreyassreenivas) [inline internal/resources/cluster/constants.go:19]: you have overwritten my changes

File: examples/data-sources/tmc_cluster_group/data-source.tf [hunk 6]
```diff
@@ -1,4 +1,4 @@
 # Read Tanzu Mission Control cluster group : fetch cluster group details
-data "tmc_cluster_group" "read_cluster_group" {
+data "tanzu-mission-control_cluster_group" "read_cluster_group" {
   name = "default"
```
Comment (shreyassreenivas) [inline examples/data-sources/tmc_cluster_group/data-source.tf:3]: Similar to data-source.tf, there is a provider.tf. Please update those files as well

File: internal/resources/cluster/constants.go (hunk 5 above)
Comment (shobha2626) [reply internal/resources/cluster/constants.go]: Yes, done

File: internal/resources/cluster/tkgvsphere/resource_tkgvsphere.go [hunk 7]
```diff
@@ -16,7 +16,7 @@ import (
 
 var TkgVsphereClusterSpec = &schema.Schema{
 	Type:        schema.TypeList,
-	Description: "The TKG Vsphere cluster spec",
+	Description: "The Tanzu Kubernetes Grid (TKGm) cluster spec",
```
[the remark this answers is not in the document]
Comment (shobha2626) [reply internal/resources/cluster/tkgvsphere/resource_tkgvsphere.go]: Done

File: examples/data-sources/tmc_cluster_group/data-source.tf (hunk 6 above)
Comment (shobha2626) [reply examples/data-sources/tmc_cluster_group/data-source.tf:3]: Done

File: docs/index.md [hunk 8]
```diff
@@ -41,7 +41,7 @@ From this page, you can generate a new API token, and then copy it to use for th
 ## Example Usage
 
 ```terraform
-provider "tmc" {
+provider "tanzu-mission-control" {
   endpoint = var.endpoint # optionally use TMC_ENDPOINT env var
   token = var.token # optionally use TMC_CSP_TOKEN env var
```
Comment (shreyassreenivas) [inline docs/index.md:46]: update the token to reflect the new attribute key value

Comment (shobha2626) [reply docs/index.md:46]: This is as part of PR https://github.com/vmware/terraform-provider-tanzu-mission-control/pull/28

File: examples/provider/provider.tf [hunk 9]
```diff
@@ -1,4 +1,4 @@
-provider "tmc" {
+provider "tanzu-mission-control" {
   endpoint = var.endpoint # optionally use TMC_ENDPOINT env var
   token = var.token # optionally use TMC_CSP_TOKEN env var
```
[the remark this answers is not in the document]
Comment (shobha2626) [reply examples/provider/provider.tf:3]: This is as part of PR https://github.com/vmware/terraform-provider-tanzu-mission-control/pull/28

**Review by shreyassreenivas — APPROVED** [review]

**Pull request merged**